### ISSUE TITLE
Username and Password into Sketch for MQTT

### ISSUE THREAD

**Issue by smartboart** [jkairys/growatt-esp8266#2]
Hi,
I am using mqtt server with username and password.
Can you please include mqtt username,pass option in your sketch?

Thanks in advance,

**Comment by smartboart** [comment]
Think i got it with this 3 lines...
Can you confirm?

const char* mqtt_username = "XXXX";
const char* mqtt_password = "XXXXXXX";

 if (client.connect("GrowattInverter",  mqtt_username, mqtt_password)) {

**Comment by smartboart** [comment]
ok this might work....but i only get 2 states over mqtt...- inverter error showing e2 and inverter status showing init...ihave the growatt 1000s inverter...can you help ?

**Comment by jkairys** [comment]
I've updated this project to work more like a library now, so the WiFi / MQTT settings are configured separately to the inverter reading codebase. Theres an example application in `examples/growatt-mqtt` that should explain how to use it. I've moved the WiFi and MQTT connection handling into another library `iot-thing` which attempts to simplify my most common use cases for connecting and reconnecting to WiFi and an MQTT server.
